ui/src/containers/SampleGridTableContainer.jsx

```diff
@@ -758,15 +766,92 @@ class SampleGridTableContainer extends React.Component {
         }
         return null;
       });
-
     return sampleItemList;
   }
 
+  getPuckSampleItems(puck, sampleNumber) {
+    const sample = Object.values(this.props.sampleList).find(
+      (sample) => sample.location === puck + ':' + sampleNumber,
+    );
+    if (typeof sample === 'undefined') {
+      return null;
+    }
+    // const key = sample.sampleID;
+    const key = puck + ':' + sampleNumber;
```

Comment:
If you move this to the top of the function, you can use it inside the `find` loop.